### PR TITLE
bbc-a11y intermittent failure 

### PR DESCRIPTION
**Pull request by alex-magana**
Resolves https://github.com/bbc/simorgh-infrastructure/issues/511 and https://github.com/bbc/simorgh-infrastructure/issues/534

Depends on: https://github.com/BBC-News/devops-tools-ecr/pull/119

**Overall change:** This PR proposes using `test:prod:jenkins` to run `bbc-a11y` on Jenkins and
`test:prod:ci` for Travis owing to the Debian related error in lighthouse.

**Code changes:**

- Add `bbcA11y` and `test:puppeteer` to `test:prod:jenkins` to run bbc-a11y and puppeteer on Jenkins.
- Add `--no-sandbox` as a launch argument to puppeteer to enable running puppeteer as `root`

---

With the increased processing capacity on Jenkins, the intermittent bbcA11y script failure
has been resolved. This has been confirmed by running the bbcA11y script as part of
`test:prod:ci` in builds 3 - 7 of
https://ci.news.tools.bbc.co.uk/blue/organizations/jenkins/simorgh/activity?branch=511-bbc-a11y-test-latest-electron-check

In the process of testing the bbcA11y script, errors emerged in the `lighthouse` and `test:puppeteer`
scripts. These errors were

|script|error|
|------------|---------|
| lighthouse | Error: Protocol error (Tracing.start): 'Tracing.start' wasn't found|
| puppeteer  | Error: Failed to launch chrome! [0709/110822.721256:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported.|

The lighthouse error is due to Tracing being disabled in `chromium` by Debian packaging. This affects us since the docker image we use is Debian based as per the information below, derived from
`cat /etc/os-release`.

```
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

For more details on this error, see the links below.
https://github.com/GoogleChrome/lighthouse/issues/7246
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=922431
https://bugs.chromium.org/p/chromium/issues/detail?id=931143

The `lighthouse` script in `test:prod:ci` steps works well on Travis which upon inspection
runs on Ubuntu as per the info below.

```
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
```

The puppeteer error listed in the table above occurs due to running
puppeteer without `--no-sandbox` when running the system as `root`. To address this issue,
I've defined `--no-sandbox` args as part on the puppeteer launch config.

Deactivating sandbox is considered a possible exploitation avenue. The sandbox mode is used
by Chrome to shield the host from attacks in case malicious content is accessed from Chrome
on an agent. Since the puppeteer script runs from Travis and in a docker container on Jenkins
behind our certs. We should be relatively safe. Additionally, we use `-u root` on Jenkins because the privilege is required for steps we run.

See the links below for further details.
https://github.com/GoogleChrome/puppeteer/issues/3698
https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-in-docker

---
- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "test:ampValidate": "npm run killApp && npm run build -- --hide-modules --colors && run-p --race start amp:validate",
     "test:ampValidate:ci": "npm run build -- --hide-modules --colors && run-p --race start amp:validate",
     "test:prod:ci": "npm run start & npm install --no-save cypress@latest bbc-a11y@latest lighthouse@latest npm-run-all@latest puppeteer@latest && run-p cypress bbcA11y lighthouse test:puppeteer",
-    "test:prod:jenkins": "npm install --no-save cypress@3.3.0 npm-run-all@4.1.5 && run-p --race start cypress",
+    "test:prod:jenkins": "npm install --no-save cypress@latest bbc-a11y@latest npm-run-all@4.1.5 puppeteer@latest && run-p --race start cypress bbcA11y test:puppeteer",
```
Comment (jamesbhobbs) [inline package.json]: Why is any of this script custom? we should just be using `test:prod:ci`

**Review by jamesbhobbs — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (alex-magana) [reply package.json]: We're doing this due to the error below, with which `lighthouse` fails on Jenkins.
`Error: Protocol error (Tracing.start): 'Tracing.start' wasn't found`

The error is due to `Tracing ` being disabled on chromium via Debian packaging as captured by
https://github.com/GoogleChrome/lighthouse/issues/7246
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=922431
https://bugs.chromium.org/p/chromium/issues/detail?id=931143

It's reported to be working in `chromium version 72.0.3626.53`. I'll attempt running lighthouse by targeting the specific chromium version from the docker image we use.
The challenge I foresaw with this was locking the chromium version on a shared image hence my making the script custom.

**Review by mbzain76 — APPROVED** [review]
Ok

**Review by DemonusPC — APPROVED** [review]
Looks dank to me

**Pull request merged**